packages/orchestrator/internal/proxy/proxy.go

```diff
@@ -0,0 +1,196 @@
+package proxy
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"go.uber.org/zap"
+	"html/template"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/e2b-dev/infra/packages/shared/pkg/meters"
+	"github.com/e2b-dev/infra/packages/shared/pkg/smap"
+)
+
+//go:embed proxy_browser_502.html
+var proxyBrowser502PageHtml string
+
+var browserRegex = regexp.MustCompile(`(?i)mozilla|chrome|safari|firefox|edge|opera|msie`)
+var browserTemplate = template.Must(template.New("template").Parse(proxyBrowser502PageHtml))
+
+type htmlTemplateData struct {
+	SandboxId   string
+	SandboxHost string
+	SandboxPort string
+}
+
+type jsonTemplateData struct {
+	Error     string `json:"error"`
+	SandboxId string `json:"sandbox_id"`
```

Comment:
it was `sandboxId` before, there are already tests in sdk, if there isn't a strong argument to rename it, please revert to the old naming